packages/new-polymath-issuer/src/pages/DividendsWizard/Container.tsx

```diff
@@ -203,48 +208,118 @@ export class ContainerBase extends Component<Props, State> {
 
   public render() {
     const { securityTokenSymbol, checkpointIndex } = this.props;
+    const fetcher = createCheckpointsBySymbolFetcher({
+      securityTokenSymbol,
+    });
+
     const { step } = this.state;
     const parsedCheckpointIndex = parseInt(checkpointIndex, 10);
     return (
       <Page title="Create New Dividend Distribution">
         <DataFetcher
           watchProps={this.state}
-          fetchers={[
-            createTaxWithholdingListBySymbolAndCheckpointFetcher({
-              securityTokenSymbol,
-              checkpointIndex: parsedCheckpointIndex,
-              dividendType: DividendModuleTypes.Erc20,
-            }),
-            createCheckpointBySymbolAndIdFetcher({
-              securityTokenSymbol,
-              checkpointIndex: parsedCheckpointIndex,
-            }),
-          ]}
-          render={(
-            {
-              taxWithholdings,
-              checkpoints: [checkpoint],
-            }: {
-              taxWithholdings: types.TaxWithholdingEntity[];
-              checkpoints: types.CheckpointEntity[];
-            },
-            loading: boolean
-          ) => {
+          fetchers={[fetcher]}
+          render={(data: { checkpoints: types.CheckpointEntity[] }) => {
+            const { checkpoints } = data;
+            const fetchers = checkpoints.map(({ index }) =>
+              createDividendsByCheckpointFetcher(
+                {
+                  securityTokenSymbol,
+                  checkpointIndex: index,
+                },
+                { propKey: `${index}` }
+              )
+            );
             return (
-              <Presenter
-                createDividendDistribution={this.createDividendDistribution}
-                updateTaxWithholdingList={this.updateTaxWithholdingList}
-                stepIndex={step}
-                securityTokenSymbol={securityTokenSymbol}
-                checkpoint={checkpoint}
-                onNextStep={this.nextStep}
-                onPreviousStep={this.previousStep}
-                taxWithholdings={taxWithholdings}
-                downloadTaxWithholdingList={this.downloadTaxWithholdingList}
-                downloadSampleExclusionList={this.downloadSampleExclusionList}
-                fetchBalance={this.fetchBalance}
-                fetchIsValidToken={this.fetchIsValidToken}
-                isLoadingData={loading}
+              <DataFetcher
+                watchProps={this.state}
+                fetchers={fetchers}
+                render={(dividendsData: {
+                  [key: string]: types.DividendEntity[];
+                }) => {
+                  const allDividendsCompleted = every(
```

Comment:
For readability, I'd suggest avoiding this level of nesting (considering that this is already a DataFetcher within another DataFetcher). I'd flatten the array first, store it in a variable, then call `every` on that variable and simply return the `completed` calculation in the callback

Reply:
I tried to do it without 2 nested `map` statements, but could not, because we have an object that has multiple arrays, each array has an array of investors, so the data is actually tucked in the third level.

Reply:
`flatten` should take care of that by itself without the need for `map`